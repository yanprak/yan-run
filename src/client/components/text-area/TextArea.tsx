import React, { FC } from 'react';
import { OwnProps } from './types';
import './text-area.scss';

const TextArea: FC<OwnProps> = (props: OwnProps) => {
  const {
    resizable,
    children,
    ...otherProps
  } = props;
  const resizableClass = resizable ? 'text-area_resizable' : '';
  const className = `text-area ${resizableClass} ${props.className || ''}`;
  return (
    <textarea {...otherProps} className={className}>{children}</textarea>
  );
};

export default TextArea;
