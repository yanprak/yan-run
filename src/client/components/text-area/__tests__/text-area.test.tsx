import React from 'react';
import TestRenderer from 'react-test-renderer';
import TextArea from '../TextArea';

describe('TextArea tests', () => {
  test('Snapshot:', () => {
    const component = TestRenderer.create(
      <TextArea resizable placeholder="Text" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
