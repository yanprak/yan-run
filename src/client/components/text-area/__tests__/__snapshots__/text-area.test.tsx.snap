// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea tests Snapshot: 1`] = `
<textarea
  className="text-area text-area_resizable "
  placeholder="Text"
/>
`;
